tests/e2e-kubernetes/testsuites/cache.go

```diff
@@ -0,0 +1,237 @@
+	checkBasicFileOperations := func(ctx context.Context, pod *v1.Pod, bucketName, basePath string) {
+		framework.Logf("Checking basic file operations inside pod %s at %s", pod.UID, basePath)
+
+		dir := filepath.Join(basePath, "test-dir")
+		first := filepath.Join(basePath, "first")
+		second := filepath.Join(dir, "second")
+
+		seed := time.Now().UTC().UnixNano()
+		testWriteSize := 1024 // 1KB
+
+		checkWriteToPath(f, pod, first, testWriteSize, seed)
+		checkListingPathWithEntries(f, pod, basePath, []string{"first"})
+		// Test reading multiple times to ensure cached-read works
+		for i := 0; i < 3; i++ {
+			checkReadFromPath(f, pod, first, testWriteSize, seed)
+		}
+
+		// Now remove the file from S3
+		deleteObjectFromS3(ctx, bucketName, "first")
+
+		// Ensure the data still read from the cache - without cache this would fail as its removed from underlying bucket
+		checkReadFromPath(f, pod, first, testWriteSize, seed)
+
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("mkdir %s && cd %s && echo 'second!' > %s", dir, dir, second))
+		e2evolume.VerifyExecInPodSucceed(f, pod, fmt.Sprintf("cat %s | grep -q 'second!'", second))
+		checkListingPathWithEntries(f, pod, dir, []string{"second"})
+		checkListingPathWithEntries(f, pod, basePath, []string{"test-dir"})
+		checkDeletingPath(f, pod, first)
+		checkDeletingPath(f, pod, second)
+	}
+
+	createPod := func(ctx context.Context, additionalMountOptions []string, podModifiers ...func(*v1.Pod)) (*v1.Pod, string) {
+		cacheDir := randomCacheDir()
+
+		vol := createVolumeResourceWithMountOptions(ctx, l.config, pattern, append(additionalMountOptions, fmt.Sprintf("cache %s", cacheDir)))
+		deferCleanup(vol.CleanupResource)
+
+		bucketName := bucketNameFromVolumeResource(vol)
+
+		pod := e2epod.MakePod(f.Namespace.Name, nil, []*v1.PersistentVolumeClaim{vol.Pvc}, admissionapi.LevelBaseline, "")
+		ensureCacheDirExistsInNode(pod, cacheDir)
+		for _, pm := range podModifiers {
+			pm(pod)
+		}
+
+		pod, err := createPod(ctx, f.ClientSet, f.Namespace.Name, pod)
+		framework.ExpectNoError(err)
+		deferCleanup(func(ctx context.Context) error { return e2epod.DeletePodWithWait(ctx, f.ClientSet, pod) })
+
+		return pod, bucketName
+	}
+
+	Describe("Cache", func() {
+		Describe("Local", func() {
+			It("basic file operations as root", func(ctx context.Context) {
+				pod, bucketName := createPod(ctx, []string{"allow-delete"}, func(pod *v1.Pod) {
+					pod.Spec.Containers[0].SecurityContext.RunAsUser = ptr.To(root)
+					pod.Spec.Containers[0].SecurityContext.RunAsGroup = ptr.To(root)
+				})
+				checkBasicFileOperations(ctx, pod, bucketName, e2epod.VolumeMountPath1)
+			})
+
+			It("basic file operations as non-root", func(ctx context.Context) {
+				mountOptions := []string{
+					"allow-delete",
+					"allow-other",
+					fmt.Sprintf("uid=%d", *e2epod.GetDefaultNonRootUser()),
+					fmt.Sprintf("gid=%d", defaultNonRootGroup),
```

Comment:
Nit: Would these constants be useful as functions/moved to `e2epod`?

Reply:
`e2epod` is a Kubernetes package, but I can move these constants to `util.go` in a follow-up PR.